src/sidebar/test/websocket-test.js

```diff
@@ -28,49 +32,57 @@ describe('websocket wrapper', function() {
     console.warn.restore();
   });
 
-  it('should reconnect after an abnormal disconnection', function() {
-    new Socket('ws://test:1234');
-    assert.ok(fakeSocket);
-    const initialSocket = fakeSocket;
-    fakeSocket.onopen({});
-    fakeSocket.onclose({ code: 1006 });
-    clock.tick(2000);
-    assert.ok(fakeSocket);
-    assert.notEqual(fakeSocket, initialSocket);
-  });
+  context('when the connection is closed by the browser or server', () => {
+    it('should reconnect after an abnormal disconnection', function() {
+      new Socket('ws://test:1234');
+      assert.ok(fakeSocket);
+      const initialSocket = fakeSocket;
+      fakeSocket.onopen({});
+      fakeSocket.onclose({ code: CLOSE_ABNORMAL });
+      clock.tick(2000);
+      assert.ok(fakeSocket);
+      assert.notEqual(fakeSocket, initialSocket);
+    });
 
-  it('should reconnect if initial connection fails', function() {
-    new Socket('ws://test:1234');
-    assert.ok(fakeSocket);
-    const initialSocket = fakeSocket;
-    fakeSocket.onopen({});
-    fakeSocket.onclose({ code: 1006 });
-    clock.tick(4000);
-    assert.ok(fakeSocket);
-    assert.notEqual(fakeSocket, initialSocket);
-  });
+    it('should reconnect if initial connection fails', function() {
+      new Socket('ws://test:1234');
+      assert.ok(fakeSocket);
+      const initialSocket = fakeSocket;
+      fakeSocket.onopen({});
+      fakeSocket.onclose({ code: CLOSE_ABNORMAL });
+      clock.tick(4000);
+      assert.ok(fakeSocket);
+      assert.notEqual(fakeSocket, initialSocket);
+    });
 
-  it('should send queued messages after a reconnect', function() {
-    // simulate WebSocket setup and initial connection
-    const socket = new Socket('ws://test:1234');
-    fakeSocket.onopen({});
+    it('should send queued messages after a reconnect', function() {
+      // simulate WebSocket setup and initial connection
+      const socket = new Socket('ws://test:1234');
+      fakeSocket.onopen({});
 
-    // simulate abnormal disconnection
-    fakeSocket.onclose({ code: 1006 });
+      // simulate abnormal disconnection
+      fakeSocket.onclose({ code: CLOSE_ABNORMAL });
 
-    // enqueue a message and check that it is sent after the WS reconnects
-    socket.send({ aKey: 'aValue' });
-    fakeSocket.onopen({});
-    assert.calledWith(fakeSocket.send, '{"aKey":"aValue"}');
-  });
+      // enqueue a message and check that it is sent after the WS reconnects
+      socket.send({ aKey: 'aValue' });
+      fakeSocket.onopen({});
+      assert.calledWith(fakeSocket.send, '{"aKey":"aValue"}');
+    });
 
-  it('should not reconnect after a normal disconnection', function() {
-    const socket = new Socket('ws://test:1234');
-    socket.close();
-    assert.called(fakeSocket.close);
-    const initialSocket = fakeSocket;
-    clock.tick(2000);
-    assert.equal(fakeSocket, initialSocket);
+    [CLOSE_NORMAL, CLOSE_GOING_AWAY].forEach(closeCode => {
```

Comment:
This is the one functional change here. This test is now parameterized by close status code.